include/detail/aglShaderHolder.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <container/seadPtrArray.h>
+#include <heap/seadDisposer.h>
+#include <hostio/seadHostIOReflexible.h>
+#include <prim/seadSafeString.h>
+#include <resource/seadArchiveRes.h>
+#include "agl/common/aglShaderProgramArchive.h"
+
+namespace agl {
+
+class ShaderProgram;
+
+namespace detail {
+
+class ShaderHolder {
+    SEAD_SINGLETON_DISPOSER(ShaderHolder);
+
+public:
+    ShaderHolder();
+    virtual ~ShaderHolder();
+
+    void initialize(sead::ArchiveRes* archiveRes, sead::Heap* heap);
+    sead::FormatFixedSafeString<1024> genMessage(sead::hostio::Context* context);
+    void listenPropertyEvent(const sead::hostio::PropertyEvent* property_event);
+
+private:
+    sead::FixedPtrArray<ShaderProgram, 206> mShaderPrograms;
+    ShaderProgramArchive mShaderProgramArchives[6];
+    char mArchiveOptions;
```

Comment:
Are you sure this isn't just a boolean?

Reply:
That could still be a bool from my understanding

Reply:
Might be - no idea, and I don't find enough use of this in the whole game. It's initialized to `1`, then only ever read with this instruction, passed as `uint` into a function which then only reads the `& 4` bit, ignoring everything else.

Reply:
AFAIK it's a boolean on Wii U. I'll double check later, but it's a boolean in my repo as well.

Reply:
chatted with aboood, it works as a bool and should be changed for the PR

Reply:
Adjusted accordingly.